core/app/models/spree/shipment.rb

```diff
@@ -82,7 +82,7 @@ def can_transition_from_pending_to_shipped?
 
     def can_transition_from_pending_to_ready?
       order.can_ship? &&
-        inventory_units.all? { |iu| iu.allow_ship? || iu.canceled? } &&
+        inventory_units.on_hand.all? { |iu| iu.allow_ship? || iu.canceled? } &&
```

Comment:
I think that this won't work if the current shipment only has inventory units in a different state, eg `backordered`. In this case `all?` returns true and it will incorrectly move the shipment to ready

Reply:
Thats what I though too but we actually used on_hand inventory units only because we saw this method from inventory_unit.rb:

```
def allow_ship?
     on_hand?
   end
```

Reply:
I saw and fixed some rspec does not break by this code

Reply:
I think that with `InventoryUnit.pre_shipment` we'll also cover `backorders` inventory units

Reply:
So we should change the `def allow_ship?` method

Reply:
No, I meant changing `inventory_units.on_hand.all? {}` into `inventory_units.pre_shipment.all? {}` but I'm still not sure we'll cover all scenarios since we'll filter out all `canceled` inventory units (making the `|| iu.canceled?` check in the block useless). I need to understand better what that code does, and which kind of inventory units we expect to have when the shipment is ready.

Reply:
So I wait your feedback before to add some tests?

Reply:
yes, please let me (and/or other users as well) think a bit about this, thanks again!